invisible_cities/reco/icaro_components_test.py

```diff
@@ -0,0 +1,98 @@
+import numpy         as np
+import numpy.testing as npt
+import pandas        as pd
+
+from hypothesis.strategies import integers
+from hypothesis.strategies import floats
+from hypothesis            import given
+from pytest                import mark
+from pytest                import raises
+
+from   sklearn.linear_model import RANSACRegressor
+
+from .. core.testing_utils  import assert_dataframes_equal
+from .. core                import core_functions   as core
+
+from .                      import icaro_components as icarcomp
+
+
+@mark.parametrize("signal", (icarcomp.type_of_signal.nS1, icarcomp.type_of_signal.nS2))
+@given(nsignals= integers(min_value = 1,
+                          max_value = 1e4))
+def test_selection_nS_mask_and_checking_right_output(nsignals, signal):
+    nevt = int(1e4)
+    data = np.concatenate([np.zeros(nevt- nsignals), np.ones(nsignals)])
+    np.random.shuffle(data)
+    data = pd.DataFrame({'nS1': data, 'nS2': data, 'event': range(nevt)})
+
+    mask = icarcomp.selection_nS_mask_and_checking(data, signal)
+
+    assert np.sum(mask) == nsignals
+
+
+@given(integers(min_value = 1,
+                max_value = 1e4))
+def test_selection_nS_mask_and_checking_consistency(nsignals):
+    nevt = int(1e4)
+    data = np.concatenate([np.zeros(nevt- nsignals),
+                          np.ones  (nsignals)])
+    np.random.shuffle(data)
+
+    data = pd.DataFrame({'nS1': data, 'event': range(nevt)})
+
+    mask    = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS1)
+    mask_re = icarcomp.selection_nS_mask_and_checking(data, icarcomp.type_of_signal.nS1, input_mask=mask)
+    npt.assert_equal(mask, mask_re)
+
+
+@given(integers(min_value = 1,
+                max_value = 1e4),
+       integers(min_value = 1,
+                max_value = 1e4))
+def test_selection_nS_mask_and_checking_concatenating(ns1, ns2):
+    nevt = int(1e4)
+    dataS1 = np.concatenate([np.zeros(nevt- ns1),
+                             np.ones (ns1)])
+    dataS2 = np.concatenate([np.zeros(nevt- ns2),
+                             np.ones (ns2)])
+    np.random.shuffle(dataS1)
+    np.random.shuffle(dataS2)
```

Comment:
hypothesis & random

Reply:
I don't necessarily agree with this one. Although it's true random are used, it is still deterministic, it's only for ordering. Am I missing something?

Reply:
As discussed with peras, while in this particular case it doesn't matter because the function doesn't care about the order, it is not a good practice. Also, hypothesis can be heavy on runtime, so it's better reserved for cases where it is more useful.